File: PYTHON/PROGRAM/Komputer.py
class Komputer:
    def __init__(self, nama="", cpu=None, ram_list=None, harddrive=None, memory=None):
        self.__nama = nama
        self.__cpu = cpu 
        self.__ram_list = ram_list if ram_list is not None else []
        self.__harddrive = harddrive
        self.__memory = memory

    def add_ram(self, ram):
        """Menambahkan RAM ke dalam komputer."""
        self.__ram_list.append(ram)

    def get_nama(self):
        return self.__nama

    def get_cpu(self):
        return self.__cpu

    def get_ram_list(self):
        return self.__ram_list

    def get_harddrive(self):
        return self.__harddrive
    
    def get_memory(self):
        return self.__memory
    
    def get_total_ram_capacity(self):
        """Menghitung total kapasitas RAM yang terpasang."""
        total = 0
        for ram in self.__ram_list:
            total += ram.get_kapasitas_gb()
        return total
    
    def getTotalRamCapacity(self):
        """Menghitung total kapasitas RAM yang terpasang (metode alternatif)."""
        total = 0
        for ram in self.__ram_list:
            total += ram.get_kapasitas_gb()
        return total

    def display_info(self):
        print("Informasi Komputer:")
        print(f"Nama      : {self.__nama}")
        
        if self.__cpu:
            print(f"CPU       : {self.__cpu.get_merk()} {self.__cpu.get_nama()} "
                  f"({self.__cpu.get_jumlah_core()} Core) ~{self.__cpu.get_kecepatan_ghz()}GHz")
        
        print("RAM List  : ")
        for i, ram in enumerate(self.__ram_list):
            print(f"  [{i+1}] {ram.get_merk()} {ram.get_nama()} "
                  f"({ram.get_kapasitas_gb()} GB) {ram.get_ddr()}")
        
        if self.__harddrive:
            print(f"Harddrive : {self.__harddrive.get_tipe_drive()} {self.__harddrive.get_merk()} "
                  f"{self.__harddrive.get_nama()} ({self.__harddrive.get_kapasitas_gb()} GB)")
        
        if self.__memory:
            print(f"Memory    : {self.__memory.get_jenis_memory()} {self.__memory.get_merk()} "
                  f"{self.__memory.get_nama()} ({self.__memory.get_kapasitas_gb()} GB) "
                  f"{self.__memory.get_tipe_drive()} | Kecepatan Transfer: "
                  f"{self.__memory.get_kecepatan_transfer()} MB/s")

File: JAVA/PROGRAM/Komputer.java
import java.util.ArrayList;
import java.util.List;

public class Komputer {
    private String nama;
    private Cpu cpu;
    private List<Ram> ramList;
    private Harddrive harddrive;
    private Memory memory;
    
    // Konstruktor
    public Komputer() {
        this.nama = "";
        this.cpu = null;
        this.ramList = new ArrayList<>();
        this.harddrive = null;
        this.memory = null;
    }
    
    public Komputer(String nama, Cpu cpu, List<Ram> ramList, Harddrive harddrive) {
        this.nama = nama;
        this.cpu = cpu;
        this.ramList = ramList != null ? ramList : new ArrayList<>();
        this.harddrive = harddrive;
        this.memory = null;
    }
    
    public Komputer(String nama, Cpu cpu, List<Ram> ramList, Memory memory) {
        this.nama = nama;
        this.cpu = cpu;
        this.ramList = ramList != null ? ramList : new ArrayList<>();
        this.harddrive = null;
        this.memory = memory;
    }
    
    // Metode untuk menambahkan RAM
    public void addRam(Ram ram) {
        ramList.add(ram);
    }
    
    // Getter dan Setter
    public String getNama() {
        return nama;
    }
    
    public void setNama(String nama) {
        this.nama = nama;
    }
    
    public Cpu getCpu() {
        return cpu;
    }
    
    public void setCpu(Cpu cpu) {
        this.cpu = cpu;
    }
    
    public List<Ram> getRamList() {
        return ramList;
    }
    
    public void setRamList(List<Ram> ramList) {
        this.ramList = ramList;
    }
    
    public Harddrive getHarddrive() {
        return harddrive;
    }
    
    public void setHarddrive(Harddrive harddrive) {
        this.harddrive = harddrive;
    }
    
    public Memory getMemory() {
        return memory;
    }
    
    public void setMemory(Memory memory) {
        this.memory = memory;
    }
    
    // Metode untuk menghitung total kapasitas RAM
    public int getTotalRamCapacity() {
        int total = 0;
        for (Ram ram : this.ramList) {
            total += ram.getKapasitasGb();
        }
        return total;
    }
    
    // Metode untuk menampilkan informasi komputer
    public void displayInfo() {
        System.out.println("Informasi Komputer:");
        System.out.println("Nama      : " + this.nama);
        
        if (this.cpu != null) {
            System.out.println("CPU       : " + this.cpu.getMerk() + " " + this.cpu.getNama() + 
                               " (" + this.cpu.getJumlahCore() + " Core) ~" + 
                               this.cpu.getKecepatanGhz() + "GHz");
        }
        
        System.out.println("RAM List  : ");
        for (int i = 0; i < this.ramList.size(); i++) {
            Ram ram = this.ramList.get(i);
            System.out.println("  [" + (i+1) + "] " + ram.getMerk() + " " + ram.getNama() + 
                               " (" + ram.getKapasitasGb() + " GB) " + ram.getDdr());
        }
        
        if (this.harddrive != null) {
            System.out.println("Harddrive : " + this.harddrive.getTipeDrive() + " " + 
                               this.harddrive.getMerk() + " " + this.harddrive.getNama() + 
                               " (" + this.harddrive.getKapasitasGb() + " GB)");
        }
        
        if (this.memory != null) {
            System.out.println("Memory    : " + this.memory.getJenisMemory() + " " + 
                               this.memory.getMerk() + " " + this.memory.getNama() + 
                               " (" + this.memory.getKapasitasGb() + " GB) " + 
                               this.memory.getTipeDrive() + " | Kecepatan Transfer: " + 
                               this.memory.getKecepatanTransfer() + " MB/s");
        }
    }
}